app/goal_routes.py

```diff
@@ -0,0 +1,97 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, jsonify, make_response, request, abort
+from .routes_helper import validate_model
+
+goals_bp = Blueprint("goals", __name__, url_prefix = "/goals")
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return make_response(jsonify({"details": "Invalid data"}), 400)
+    # Create a new goal in the database
+    new_goal = Goal(
+        title=request_body["title"]
+    )
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    response_body = dict(goal = new_goal.to_dict())
+
+    return jsonify(response_body), 201
+
+@goals_bp.route("", methods=["GET"])
+def get_all_goals():
+
+    goals = Goal.query.all()
+    goal_list = [goal.to_dict() for goal in goals]
```

Comment:
Nice job incorporating list comprehension. Note that goal_list is a list of Nonetype objects because currently to_dict does not return anything (default return value when a value is not specified is None in Python)